graph/src/data/store/mod.rs

```diff
@@ -609,6 +609,33 @@ impl CacheWeight for Entity {
     }
 }
 
+pub type Vid = Option<NonZeroU64>;
+
+#[derive(Default, Debug)]
+pub struct EntityVersion {
+    pub data: Entity,
+    /// The `vid` of the entity if it exists in the store
+    pub vid: Vid,
+}
+
+impl EntityVersion {
+    pub fn new(data: Entity, vid: Vid) -> Self {
+        EntityVersion { data, vid }
+    }
+}
+
+impl From<EntityVersion> for Entity {
+    fn from(ev: EntityVersion) -> Self {
+        ev.data
+    }
+}
+
+impl From<Entity> for EntityVersion {
+    fn from(data: Entity) -> Self {
+        Self { data, vid: None }
+    }
+}
```

Comment:
Is this impl necessary or useful? It could be error prone to implicitly set the vid to None.